backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InitializersLowering.kt

```diff
@@ -84,7 +84,7 @@ internal class InitializersLowering(val context: CommonBackendContext) : ClassLo
                     // We shall keep initializer for constants for compile-time instantiation.
                     declaration.initializer =
                             if (initExpression is IrConst<*> &&
-                                    (initExpression.type.isPrimitiveType() || initExpression.type.isString())) {
+                                    declaration.correspondingPropertySymbol?.owner?.isConst == true) {
```

Comment:
This fix [still](https://github.com/JetBrains/kotlin-native/pull/4570#discussion_r536036700) relies on the fact that if the property is `const`, then its initializer is `IrConst`.
Imagine what would happen if this wasn't true. If we had a `const` property with non-`IrConst` initializer, this code wouldn't keep the `declaration.initializer` in its place. Then https://github.com/JetBrains/kotlin/blob/b81ec2c8f05df39cfb7918d7c2032f0a1c8c6df9/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/optimizations/PropertyAccessorInlineLowering.kt#L51 would break.

But now I see that this way it is easier to duplicate the initializer. So let's keep it so, but please add the comment to the code.

Reply:
> If we had a const property with non-IrConst initializer

Is this possible? 

> But now I see that this way it is easier to duplicate the initializer. So let's keep it so, but please add the comment to the code.

So we should try to delete `initExpression is IrConst<*>`  and take into account only the second condition?